app/assets/javascripts/initializers/initializeCommentsPage.js.erb

```diff
@@ -89,6 +89,12 @@ function initializeCommentsPage() {
           })(i);
         }
       })();
+
+      const allReplyCommentTextAreas = document.querySelectorAll('[id^="textarea"]');
+      for (var i = 0; i < allReplyCommentTextAreas.length; i++) {
+        const element = allReplyCommentTextAreas[i];
+        element.addEventListener('focusin', () => initializeMentionAutocomplete(element));
```

Comment:
Won't this keep adding `focusin` event listeners each time a new comment is added?

It's probably a micro optimization but let's say I'm reading a long thread:

- I add a comment, this runs and attaches 1 event listener to the present comment textarea
- I scroll a bit, this runs and it attaches 1 event listener to the previous comment textarea and to the one I'm in as well

Are textarea nodes removed when a comment is published or dismissed?

Mostly wondering if having a global `focusin` handler on the container wouldn't be better, if the event can be bubbled up obviously

Reply:
yeah you're right - not sure why I didn't process that was happening 😄  I'll take a look into it - definitely don't want endless listeners attached!

Reply:
Looking at it, this is only going to have an impact if a user clicks 'reply' on two or more comments without adding their comment, as the textarea nodes are dynamically appended/removed from the DOM. 

It's still not great though, and we can get around it by not using an anonymous function as the listener (adding the same event listener twice has no effect (https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#multiple_identical_event_listeners) 🙂